packages/core/content-releases/server/src/controllers/release.ts

```diff
@@ -1,88 +1,137 @@
-      const hasEntryAttached: GetContentTypeEntryReleases.Request['query']['hasEntryAttached'] =
-        typeof query.hasEntryAttached === 'string' ? JSON.parse(query.hasEntryAttached) : false;
+    await validatefindByDocumentAttachedParams(query);
 
-      const data = hasEntryAttached
-        ? await releaseService.findManyWithContentTypeEntryAttached(contentTypeUid, entryId)
-        : await releaseService.findManyWithoutContentTypeEntryAttached(contentTypeUid, entryId);
+    const { contentType, entryDocumentId, hasEntryAttached, locale } = query;
+    const isEntryAttached =
+      typeof hasEntryAttached === 'string' ? Boolean(JSON.parse(hasEntryAttached)) : false;
 
-      ctx.body = { data };
+    if (isEntryAttached) {
+      const releases = await releaseService.findMany({
+        where: {
+          releasedAt: null,
+          actions: {
+            contentType,
+            entryDocumentId: entryDocumentId ?? null,
+            locale: locale ?? null,
+          },
+        },
+        populate: {
+          actions: {
+            fields: ['type'],
+          },
+        },
+      });
+      ctx.body = { data: releases };
     } else {
-      const query: GetReleases.Request['query'] = await permissionsManager.sanitizeQuery(ctx.query);
-      const { results, pagination } = await releaseService.findPage(query);
-
-      const data = results.map((release: ReleaseWithPopulatedActions) => {
-        const { actions, ...releaseData } = release;
-
-        return {
-          ...releaseData,
+      const relatedReleases = await releaseService.findMany({
+        where: {
+          releasedAt: null,
           actions: {
-            meta: {
-              count: actions.count,
-            },
+            contentType,
+            entryDocumentId: entryDocumentId ?? null,
+            locale: locale ?? null,
           },
-        };
+        },
       });
 
-      const pendingReleasesCount = await strapi.db.query(RELEASE_MODEL_UID).count({
+      const releases = await releaseService.findMany({
         where: {
+          $or: [
+            {
+              id: {
+                $notIn: relatedReleases.map((release: any) => release.id),
+              },
+            },
+            {
+              actions: null,
+            },
+          ],
           releasedAt: null,
         },
       });
-
-      ctx.body = { data, meta: { pagination, pendingReleasesCount } };
+      ctx.body = { data: releases };
     }
   },
 
+  async findPage(ctx: Koa.Context) {
+    const permissionsManager = strapi.service('admin::permission').createPermissionsManager({
+      ability: ctx.state.userAbility,
+      model: RELEASE_MODEL_UID,
+    });
+
+    await permissionsManager.validateQuery(ctx.query);
+
+    const releaseService = getService('release', { strapi });
+
+    const query: GetReleases.Request['query'] = await permissionsManager.sanitizeQuery(ctx.query);
+    const { results, pagination } = await releaseService.findPage(query);
```

Comment:
shoulndt this endpoint  also need a yup validation?

Reply:
I don't think is necessary because this query doesn't handle any custom logic and the ctx.query is already sanitized with the permissionsManager, I tried some basis cases and it's working